Plutarch/List.hs

```diff
@@ -363,3 +370,134 @@ plistEquals =
         )
         (pelimList (\_ _ -> pconstant False) (pconstant True) ylist)
         xlist
+
+preverse :: (PIsListLike l a) => Term s (l a :--> l a)
+preverse = phoistAcyclic $
+  plam $ \xs -> pfoldl # plam (\ys y -> pcons # y # ys) # pnil # xs
+
+pelemAt :: (PIsListLike l a) => Term s (PInteger :--> l a :--> a)
+pelemAt = phoistAcyclic $
+  plam $ \n xs ->
+    pif
+      (n #< 0)
+      perror
+      (pelemAt' # n # xs)
+
+pelemAt' :: (PIsListLike l a) => Term s (PInteger :--> l a :--> a)
+pelemAt' = phoistAcyclic $
+  pfix #$ plam $ \self n xs ->
+    pif
+      (n #== 0)
+      (phead # xs)
+      (self # (n - 1) #$ ptail # xs)
+
+pfind :: (PIsListLike l a) => Term s ((a :--> PBool) :--> l a :--> PMaybe a)
+pfind = phoistAcyclic $
+  pfix #$ plam $ \self f xs ->
+    pelimList
+      ( \y ys ->
+          pif
+            (f # y)
+            (pcon $ PJust y)
+            (self # f # ys)
+      )
+      (pcon PNothing)
+      xs
+
+psort :: POrd a => Term s (PList a :--> PList a)
+psort = timSort
+
+-- TODO: decide on default sort
+-- probably by benchmarking them
+
+mergesort :: POrd a => Term s (PList a :--> PList a)
```

Comment:
Naming

Reply:
Do you want all functions name psomeFunctionName or just exposed ones?

Reply:
Any export that is a Plutarch function or type from any module should be prefixed with `p`.